core/src/main/java/org/apache/calcite/util/format/postgresql/PostgresqlDateTimeFormatter.java

```diff
@@ -142,6 +227,10 @@ public class PostgresqlDateTimeFormatter {
           },
           "YY"),
       new NumberFormatPattern(
+          ChronoUnitEnum.YEARS_IN_CENTURY,
+          0,
+          Integer.MAX_VALUE,
```

Comment:
1000?

Reply:
Turns out that PostgreSQL does not limit this. It will accept 4 digit numbers.